web/src/main.js

```diff
@@ -1240,20 +1244,31 @@ Object.values(HistoryState.querystringMappings).forEach((k) => {
     HistoryState.updateHistoryFromState(state.application);
   });
 });
-// Observe for datetimechanes
+// Observe for datetime chanes
 observe(state => state.data.grid, (grid) => {
   if (grid && grid.datetime) {
     setLastUpdated();
   }
 });
-
 // Observe for legend visibility change
-observe(state => state.application.legendVisible, (legendVisible, _) => {
+observe(state => state.application.legendVisible, (legendVisible) => {
   d3.selectAll('.floating-legend').classed('mobile-collapsed', !legendVisible);
   d3.select('.floating-legend-container').classed('mobile-collapsed', !legendVisible);
   d3.select('.toggle-legend-button.up').classed('visible', !legendVisible);
   d3.select('.toggle-legend-button.down').classed('visible', legendVisible);
 });
+// Observe for left panel collapse
+observe(state => state.application.isLeftPanelCollapsed, (isLeftPanelCollapsed) => {
+  d3.select('.left-panel')
```

Comment:
For consistency and readability, I think this logic should be in its own method. :)